src/core/public/chrome/ui/header/header.tsx

```diff
@@ -211,7 +211,11 @@ export class Header extends Component<HeaderProps, State> {
     );
     const navId = htmlIdGenerator()();
     return (
-      <header className={className} data-test-subj="headerGlobalNav">
+      <header
+        className={className}
+        data-test-subj="headerGlobalNav"
+        data-disable-core-navigation="true"
```

Comment:
Would we need this if we removed the custom handling in nav_link.tsx?

Reply:
So...

I initially just removed the custom `onClick` handler in `NavLink` instead of adding the disabling property here, however there is a subtle difference of behavior between their handler and ours: Ours ignores links pointing to the currently displayed app (to let apps handles their intra-links themselves, to use react-router `Link` for example), the one for the navlinks don't.

The consequence of removing the navlink handler and using ours instead would be:

You are on the `dashboard` (NP) app. You click on the `dashboard` navlink. Our handler considers you already are on this app, and ignores the event. A full page refresh is triggered instead of `navigatingTo` the dashboard app's home.

This behavior kinda makes sense with `chrome` ui components: as they are not really owned by the displayed app, the behavior is different that what we are trying to introduce with this global handler, which is supposed to only catch events from links within the current app's dom.

One option would be to add another option attribute, such as `data-include-current-app-in-core-nav` to let the handler identify that some parts of the dom are not in the current app. AFAIK this behavior is only going to be needed on chrome components so this seems overkill.

One other option, probably slightly  better, would be to only listen to events fired from within the `AppWrapper` (`src/core/public/rendering/rendering_service.tsx`) component. The lib I'm using do not allow to bind the handler on another element than `body`, but we could just check if any parent of the event's currentTarget is our `AppWrapper` as we are already doing with the `data-disable-core-navigation` attribute. This option would still requires to keep any handler in chrome components though, it just mean we don't need to manually disable the behavior on the header and navbar.

Reply:
Can we document some of this in the code? Seems like the type of thing that will make someone want to "fix" down the road and this context could help speed up investigating.